### PR TITLE
[IRGen][runtime] Reduce object header to 8 bytes on 32-bit platforms.

### PR DESCRIPTION
**Pull request by gparker42**
SR-4353, rdar://29765800

### CODE REVIEW

**Comment by gparker42** [comment]
@swift-ci please validation test

**Comment by gparker42** [comment]
@swift-ci please test

**Comment by gparker42** [comment]
@swift-ci please test

**Comment by gparker42** [comment]
@swift-ci please test

File: lib/IRGen/GenDecl.cpp [hunk 1]
```diff
@@ -3152,7 +3151,7 @@ IRGenModule::getAddrOfGlobalConstantString(StringRef utf8) {
 
   // FIXME: Big endian-ness.
```
Comment (slavapestov) [inline lib/IRGen/GenDecl.cpp]: You can't break all those 32-bit big endian platforms... we still care about m68k right?

Comment (gparker42) [reply lib/IRGen/GenDecl.cpp]: If it's not in the tests it doesn't exist.

Having said that the comment no longer applies; any endianness problems would now be inside InlineRefCountBits. I'll remove it.

Comment (gparker42) [reply lib/IRGen/GenDecl.cpp]: Removed.

**Comment by gparker42** [comment]
@swift-ci please test

**Comment by shahmishal** [comment]
@swift-ci please test

**Pull request merged**